web/src/main/java/org/openmrs/web/Listener.java

```diff
@@ -287,7 +288,7 @@ private void setApplicationDataDirectory(ServletContext servletContext) {
 			OpenmrsUtil.setApplicationDataDirectory(appDataDir);
 		} else if (!"openmrs".equalsIgnoreCase(WebConstants.WEBAPP_NAME)) {
 			OpenmrsUtil.setApplicationDataDirectory(
-			    OpenmrsUtil.getApplicationDataDirectory() + File.separator + WebConstants.WEBAPP_NAME);
+				 Paths.get(OpenmrsUtil.getApplicationDataDirectory(), WebConstants.WEBAPP_NAME).toString());
```

Comment:
here you can actually use `new File(a, b).getAbsolutePath()` since you only combine 2 path elements